src/WebJobs.Script/Description/DocumentDBBindingMetadata.cs

```diff
@@ -25,9 +22,9 @@ public override void ApplyToConfig(JobHostConfigurationBuilder configBuilder)
             }
 
             DocumentDBConfiguration config = new DocumentDBConfiguration();
-            if (!string.IsNullOrEmpty(ConnectionString))
+            if (!string.IsNullOrEmpty(Connection))
             {
-                config.ConnectionString = ConnectionString;
+                config.ConnectionString = Utility.GetAppSettingOrEnvironmentValue(Connection);
             }
```

Comment:
No, because there will be no %% added to the values by the portal. The metadata value needs to be the app setting name which we resolve here.